cpp/src/io/text/multibyte_split.cu

```diff
@@ -217,9 +220,9 @@ __global__ void multibyte_split_seed_kernel(
   cutoff_offset tile_output_offset)
 {
   auto const thread_idx = blockIdx.x * blockDim.x + threadIdx.x;
-  if (thread_idx == 0) {
-    tile_multistates.set_inclusive_prefix(-1, tile_multistate_seed);
-    tile_output_offsets.set_inclusive_prefix(-1, tile_output_offset);
+  if (thread_idx < 32) {
+    tile_multistates.set_inclusive_prefix(thread_idx - 32, tile_multistate_seed);
+    tile_output_offsets.set_inclusive_prefix(thread_idx - 32, tile_output_offset);
```

Comment:
32 is a magic number. I do think CUDA has a `constexpr` somewhere for the warp size that you should use.

Second, by definition the first argument (tile index) to `set_inclusive_prefix` will always be negative, resulting in a negative offset added to a pointer if `num_tiles` is also less than 32, IIUC.  Is that what is intended? Is `num_tiles > 32` a precondition for this code? If so, I think that should be documented and / or asserted.

Reply:
I couldn't find one, but it seems like a sensible constant to have. The tiles are arranged as a circular array, so negative indices wrap around. `num_tiles > 32` is assumed, I'll add an assertion for it.

Reply:
https://github.com/rapidsai/cudf/blob/bf2c75113b2e0f146a5f5e8df22c244d5a39bdf3/cpp/include/cudf/detail/utilities/cuda.cuh#L36

Reply:
thanks! I grepped for `= 32` only, there are just too many ways to initialize variables 😄